src/App.jsx

```diff
@@ -14,45 +23,142 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+let previousGoods = [...goodsFromServer];
+
+const getSortedGoods = (goods, sortMethod, isReverse) => {
+  let newGoods = [...goods];
+  const isReverseSwitched = isReverse !== previousReverse;
+
+  if (isReverseSwitched) {
+    previousReverse = isReverse;
+  }
+
+  switch (sortMethod) {
+    case SORT_ALPHABETICALLY:
+      newGoods = newGoods.sort((good1, good2) => (isReverse
+        ? good2.localeCompare(good1)
+        : good1.localeCompare(good2)));
+      break;
+    case SORT_BY_LENGTH:
+      newGoods = isReverse
+        ? newGoods.sort()
+          .sort((good1, good2) => good1.length - good2.length).reverse()
+        : newGoods.sort()
+          .sort((good1, good2) => good1.length - good2.length);
```

Comment:
You can just reverse the array in the end instead

Reply:
Did that